homeassistant/components/zha/sensor.py

```diff
@@ -278,6 +301,53 @@ def native_unit_of_measurement(self) -> str:
         """Return Unit of measurement."""
         return self._channel.unit_of_measurement
 
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return device state attrs for battery sensors."""
+        attrs = {}
+        if self._channel.device_type is not None:
+            attrs["device_type"] = self._channel.device_type
+        status = self._channel.status
+        if status is not None:
+            attrs["status"] = str(status)[len(status.__class__.__name__) + 1 :]
+        return attrs
+
+
+@MULTI_MATCH(channel_names=CHANNEL_SMARTENERGY_METERING)
+class SmartEnergySummation(SmartEnergyMetering, id_suffix="summation_delivered"):
+    """Smart Energy Metering summation sensor."""
+
+    SENSOR_ATTR: int | str = "current_summ_delivered"
+    _device_class: str | None = DEVICE_CLASS_ENERGY
+    _state_class: str = STATE_CLASS_TOTAL_INCREASING
+
+    @property
+    def native_unit_of_measurement(self) -> str:
+        """Return Unit of measurement."""
+        uom = self._channel.cluster.get("unit_of_measure")
+        if uom == 0:
+            return ENERGY_KILO_WATT_HOUR
+        return self._channel.unit_of_measurement
+
+    def formatter(self, value: int) -> int | float:
+        """Numeric pass-through formatter."""
+        uom = self._channel.cluster.get("unit_of_measure")
+        if uom != 0:
+            return self._channel.summa_formatter(value)
+
+        # SmartEnergy cluster UOM 0 could be kW or kWh depending on context
+        cooked = float(value * self._channel.multiplier) / self._channel.divisor
+        if cooked <= 1e1:
+            return round(cooked, 3)
+
+        if cooked < 1e2:
+            return round(cooked, 2)
+
+        if cooked < 1e3:
+            return round(cooked, 1)
+
+        return round(cooked)
```

Comment:
Not sure it is a good thing to base number of decimal places on the magnitude of the value. Summation type of measurements are monotonically increasing and, over time, consumption during shorter periods will drown and become indiscernable. May be ok for demand/rate type, but should, in my opinion, be avoided for summations. I would prefer following the device specified formatting which can at least be overridden with quirk.

Reply:
I hear what you are saying and I agree the summation sensor is monotonically increasing, however my reasoning for this was:
- when my values are low, I'd like to see whether currently it is at 251W or at 258W (current UOM is kW)
- when I get to 1000000W, we stop displaying a  decimal after comma, because to me it really doesn't matter at this point whether it accumulated 1000000W or 1000010W 

Currently of all devices I have, highest reading is 5kWh. This switch was in operation for about 2.5 years, but I don't remember if it was ever reset.

Reply:
> 
> 
> I hear what you are saying and I agree the summation sensor is monotonically increasing, however my reasoning for this was:
> 
>     * when my values are low, I'd like to see whether currently it is at 251W or at 258W (current UOM is kW)
> 
>     * when I get to 1000000W, we stop displaying a  decimal after comma, because to me it really doesn't matter at this point whether it accumulated 1000000W or 1000010W
> 
> 
> Currently of all devices I have, highest reading is 5kWh. This switch was in operation for about 2.5 years, but I don't remember if it was ever reset.

@Adminiuga, with respect to instantaneous demand I do agree that a 0.001% difference is not necessarily interesting. For accumulated entities such as summation delivered the situation is different. If you make the resolution relative to the accumulated value you will, as the accumulated value continues to grow over time, lose the possibility to distinguish, first hourly variations, then daily variations ... and down the line also weekly and monthly variations. This is not desirable. There is no reason why my daily or monthly consumption would be less interesting just because I have operated the system for a long time. Thus, for summation sensors I expect the resolution to be absolute/independent of the magnitude of the value; e.g., a fixed number of decimal places.

Reply:
switched summation to Wh and fixed scale of 1 decimal place.

Reply:
... and Wh uom instead of kWh will break history for anyone who has previously been using kWh which is the standard used by electricity companies at least in large parts of Europe :-(

Reply:
what do other integrations use? deconz? zigbee2mqtt? Maybe good to use same uom as those to simplify migration too?

Reply:
It's a new feature, so for zha nobody was using Wh or kWh as of now

Reply:
Yeah, since we're converting kWh to Wh i don't think the device formatting is that relevant anymore. Alternatively it should be adjusted based on kWh to Wh conversion, but did not want to complicate it. Not yet.

 Actually, with HA I'm not positive we need to use device formatting for other UOMs
but at this point not ready to make the change.
It makes sense to use device formatting for BCD encoded values, but currently we do not support it. Because did not find a device in the wild yet 

Edit: if we use kWh, then how many digits after comma do we keep? 
Based on values supplied by existing devices?

@dmulcahey what's the largest value for current_summ_delivered attribute in your production and what is the divisor?

Reply:
true it is new for zha, but why make it more complicated than necessary for those who have had to use other means to collect the data in the past? For instance, I was using deconz before and discovered only after I migrated to zha that zha did not provide sensor for the mandatory attribute of the metering cluster. Old data is in kWh and changing to Wh will be very inconvenient. I am sure I am not the only one. Don't we want more people to use zha?

Reply:
w r t the resolution, I guess that can be addressed with a non-breaking PR at some point, but it seems important to get the uom right from day one. What is the rationale for favouring Wh over kWh?

Reply:
What's your current max raw value?
What's the divisor?
How many digits after dot should we keep for kWh? 1, 3 4?


The only reason switched to Wh for increased resolution

Reply:
I vote for kWh and 3 digits after comma. This would give resolution of 1W.

Reply:
3 digits after comma may be an acceptable compromise, but I am still not sure why the previous logic based on the formatting attribute should be abandoned. Is it not good to take device vendor considerations into account and in the same go provide a means to override/adjut by means of quirk?

The `current_summ_delivered` attribute is `uint48_t`. The `divisor` attribute is `uint24_t`.

Reply:
Will address in future PR. As mentioned previously, vendor formatting may be needed for BCD encoded values, but for binary values HA can decide what formatting to use based on uom and desired resolution.

Reply:
fair enough. Thanks!